src/arithmetic/func_desc.h

```diff
@@ -14,13 +14,21 @@
 /* AR_Func - Function pointer to an operation with an arithmetic expression */
 typedef SIValue(*AR_Func)(SIValue *argv, int argc);
 
+typedef struct AR_ExpNode AR_ExpNode;
+/* AR_Func_Free - Function pointer to a routine for freeing a function's arithmetic expression subtree. */
+typedef void (*AR_Func_Free)(AR_ExpNode *exp);
```

Comment:
I think this can be a bit more generalised, assuming an arithmetic function maintains an internal state (context)
this free function can be used to perform its clean-up, similar to our aggregation functions. this resembles the much needed work to blur to differences between aggregation functions and arithmetic functions.

Please avoid mentioning arithmetic expression subtree and change the argument from `AR_ExpNode *exp` to `void * ctx`